### PR TITLE
Currency: Added build_script and build_info for Shiny-Server[1.5.23.1030].

### PR DESCRIPTION
**Pull request by gauravpbankar13**
## Checklist

- [x] Have you checked and followed all the points mention in the [CONTRIBUTING.MD](https://github.com/ppc64le/build-scripts/blob/master/CONTRIBUTING.md)
- [x] Have you validated script on UBI 9 container
- [x] Did you run the script(s) on fresh container with `set -e` option enabled and observe success ?
- [x] Did you have **Legal approvals** for patch files ?

### CODE REVIEW

**Comment by gauravpbankar13** [comment]
All test cases are passing for ppc64le, but some error logs are appearing for both x86 and ppc64le.
e.g. `[2024-11-06T12:32:20.197] [ERROR] shiny-server - Failed to create bookmark state directory: /blah`

**Review by sid226 — APPROVED** [review]
Approved

**Review by rashmi-ibm — APPROVED** [review]
lgtm

**Comment by rashmi-ibm** [comment]
@nikhil-kalbande -- Can you check the buildinfo.json looks appropriate to include rshiny server build on currency.

**Comment by nikhil-kalbande** [comment]
> @nikhil-kalbande -- Can you check the buildinfo.json looks appropriate to include rshiny server build on currency.

@rashmi-ibm  build_info looks ok. but I see build script added is not in expected form. Can you please make necessary changes as per checklist I shared.
Also are creating build script for ubi 9.4 .. currently currency build support ubi 9.3 as a standard

**Comment by rashmi-ibm** [comment]
@gauravpbankar13 Do the necessary changes as per suggestion from Nikhil.

**Comment by gauravpbankar13** [comment]
@vinodk99 
Can you please review it.

**Comment by vinodk99** [comment]
> @vinodk99 Can you please review it.

Changes looks good to me

**Comment by gauravpbankar13** [comment]
Hi @nikhil-kalbande :

@vinodk99  has reviewed the changes.
Can you please check once and merge the PR.

File: s/shiny-server/shiny-server_ubi_9.3.sh [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+# Install Shiny server
+if ! make install; then
+    echo "------------------$PACKAGE_NAME:install_success_but_test_fails---------------------"
+    echo "$PACKAGE_URL $PACKAGE_NAME"
+    echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | GitHub | Fail |  Install_success_but_test_Fails"
+    exit 2
+fi
+
+# Create a tarball of the installed Shiny server
+cd /root
```
Comment (nikhil-kalbande) [inline s/shiny-server/shiny-server_ubi_9.3.sh:70]: @gauravpbankar13 Lets not keep this path hardcoded. .. you can use $HOME variable or set certain variable at start of script to set the path. use it for relative movement,

File: s/shiny-server/shiny-server_ubi_9.3.sh [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+# Create startup script
+echo "Creating startup script..."
+echo 'echo -a "wsl-shiny-container-${HOSTNAME}-is-up" sh & tail -f /dev/null & wait' > ~/startup.sh
+chmod 755 ~/startup.sh
+
+
+# Copy configuration files
+mkdir -p /etc/shiny-server
+cp /root/shiny-server/config/default.config /etc/shiny-server/shiny-server.conf
```
Comment (nikhil-kalbande) [inline s/shiny-server/shiny-server_ubi_9.3.sh:81]: make path changes on each reference of /root

File: s/shiny-server/shiny-server_ubi_9.3.sh (hunk 1 above)
Comment (gauravpbankar13) [reply s/shiny-server/shiny-server_ubi_9.3.sh:70]: Ok. Initially I was having variable for the same but @vinodk99  ask me to have it hardcoded value.
Will update it.

Comment (gauravpbankar13) [reply s/shiny-server/shiny-server_ubi_9.3.sh]: Hi @nikhil-kalbande 
I have added require changes.
Can you please review it?

File: s/shiny-server/shiny-server_ubi_9.3.sh [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+#             package and/or distribution. In such case, please
+#             contact "Maintainer" of this script.
+#
+# ----------------------------------------------------------------------------
+
+PACKAGE_URL=https://github.com/rstudio/shiny-server
+PACKAGE_NAME=shiny-server
+PACKAGE_VERSION=${1:-v1.5.23.1030}
+export ARCH=$(uname -m)
+TMPWS_DIR=/root
```
[the remark this answers is not in the document]
Comment (gauravpbankar13) [reply s/shiny-server/shiny-server_ubi_9.3.sh]: Hi @nikhil-kalbande :
Updated it with TMPWS_DIR=$HOME and tested locally as well.

File: s/shiny-server/build_info.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    "package_name" : "shiny-server",
+    "github_url": "https://github.com/rstudio/shiny-server",
+    "version": "v1.5.23.1030",
+    "default_branch": "master",
+    "package_dir": "s/shiny-server",
+    "build_script": "shiny-server_ubi_9.3.sh",
+    "validate_build_script": true,
+    "use_non_root_user": false,
+    "docker_build": false,
+    "v.*.*.*.*": {
```
Comment (prachikurade1) [inline s/shiny-server/build_info.json:12]: Should be v1.*.*.*.* or v*.*.*.*.*

**Review by prachikurade1 — CHANGES_REQUESTED** [review]

File: s/shiny-server/build_info.json (hunk 4 above)
Comment (gauravpbankar13) [reply s/shiny-server/build_info.json]: Hi @prachikurade1 :
Updated code as per requirement.

**Review by prachikurade1 — APPROVED** [review]
Approved

**Pull request merged**